Enable context drilldowns for all panels

vadimkibana opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/111893

This change enables context menu URL drill-down for all Dashboard panels.

This PR also now always shows the `<TriggerPicker>`, even if there is only a single trigger to pick from, the single trigger is pre-selected and remains always selected.

This makes it clear to what trigger URL drilldown is being attached:

![image](https://user-images.githubusercontent.com/82822460/146962326-4591dd21-6ab2-463a-ae6d-6494583eff87.png)

Trigger picker now always visible in dashboard drilldown setup:

![image](https://user-images.githubusercontent.com/82822460/146961964-6a2fcfea-f206-470e-aeda-ad7977a00516.png)



### Checklist

Delete any items that are not applicable to this PR.

- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials

vadimkibana committed:
enable drilldowns for all panels

vadimkibana committed:
update test

kibanamachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/5477)
* Commit: c9c0c37567ffa7f255c14ee8f243786118f46106
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-118054/c9c0c37567ffa7f255c14ee8f243786118f46106)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/90df011d9beceedbb1c8f610b7c8b8a561d1e810) | [after](https://github.com/elastic/kibana/commit/c9c0c37567ffa7f255c14ee8f243786118f46106) | diff |
| --- | --- | --- | --- |
| `dashboardEnhanced` | 14.1KB | 13.9KB | -259.0B |



To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/5477","number":5477,"commit":"c9c0c37567ffa7f255c14ee8f243786118f46106"}],"number":5477}
buildkite-pr-comment-->

vadimkibana commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into fix-ctx-drill

ThomThomson reviewed (approved):
Code only review, changes make sense based on the issue! LGTM

Dosant commented:
It doesn't look like it works as intended, 

for example, 
I created lens metric vis and tried to create a drilldown:

issue 1: It allows to start creating "dashboard drilldown" and when I click on it it shows the error: 

![Screen Shot 2021-11-30 at 12 16 48](https://user-images.githubusercontent.com/7784120/144037965-2d142a63-b906-4a25-b817-83c48f2479ba.png)


issue 2: Go to URL isn't available with basic license, so we can't assume there is always a way to create at least one drilldown which is "Go to URL in context menu" 


![Screen Shot 2021-11-30 at 12 19 04](https://user-images.githubusercontent.com/7784120/144038139-09a2b45b-eed7-4b10-ac48-9db29fe86ac7.png)

vadimkibana commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into fix-ctx-drill

vadimkibana committed:
Merge remote-tracking branch 'upstream/main' into fix-ctx-drill

vadimkibana committed:
add context menu trigger to all triggers

vadimkibana committed:
Merge remote-tracking branch 'upstream/main' into fix-ctx-drill

vadimkibana committed:
filter out factories with premium license

vadimkibana committed:
filter action factories according to triggers

vadimkibana committed:
Merge remote-tracking branch 'upstream/main' into fix-ctx-drill

vadimkibana committed:
fix test

peteharverson committed:
[ML] Transforms: Use KibanaThemeProvider in transform plugin (#120933)

* [ML] Use KibanaThemeProvider in transform plugin

* [ML] Add missing mock

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

peteharverson committed:
[ML] Use KibanaThemeProvider in ML plugin (#120892)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

machadoum committed:
Hosts Risk Step 2 - Hosts Page - Risk Column #119734 (#120487)

* Add Host risk classification column to All hosts table

* Add cypress test to risk column on all hosts table

* Fix unit test

* Add unit test

* Add tooltip to host risk column

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mattkime committed:
[data view management] fix multiple fields w/o conflicts (#121004)

* display and allow sorting on fields with multiple types that may or may not conflict

* update tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

ersin-erdal committed:
[RAC][Refactoring] Rename alerting types in Infra (#121061)

vladpro25 committed:
Dev Tools Console: Expose the error_trace parameter for completion (#120290)

* Expose the error_trace parameter for completion

* Dev Tools Console: Expose the error_trace parameter for completion

* Dev Tools Console: Expose the error_trace parameter for completion

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

vladpro25 committed:
Change suggestions for Sampler and Diversified sampler aggregations (#119355)

* Change suggestions for Sampler and Diversified sampler aggregations

* Change suggestions for Sampler and Diversified sampler aggregations

* Change suggestions for Sampler and Diversified sampler aggregations

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

tobiasstadler committed:
[APM] Prefer host.name over host.hostname (#119737) (#119952)

machadoum committed:
Fix wrong runtime field format on alert table (#120744)

* Fix wrong runtime field format on alert table

* Fix CI

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
skip flaky suite (#104249)

mistic committed:
skip flaky suite (#120440)

mistic committed:
skip flaky suite (#119267)

mistic committed:
skip flaky suite (#117780)

MadameSheema committed:
[Security Solution] Fixes upgrade tests (#120163)

* avoids the threshold test execution when the version is not the correct

* fixes version format

* updates details to align with the new data

* changes the assertion of the reason field depending on the version

* improves the navigation to the rule

* removes flakiness

* adds missing commit

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

stratoula committed:
[Lens] Applies new time axis for area and line charts with breakdown dimension (#120891)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Dosant commented:
Tested that issues here are addressed https://github.com/elastic/kibana/pull/118054#issuecomment-982540511 👍 
LGTM, but a have a few concerns: 


Now it is possible for every visualization to create a drilldown and in worth case the experience looks like this: 

![Screen Shot 2021-12-13 at 15 46 51](https://user-images.githubusercontent.com/7784120/145834366-e00e914c-206b-4c28-adf9-f98a3acc033d.png)


1. Because we hide trigger-picker, it is not clear that user is going to create a drilldown for context-menu. . I suggest we stop hiding trigger picker and simply show a single selected trigger in this case. This will explain the user what kind of interaction are they creating
2. I suggest we fix this issue - https://github.com/elastic/kibana/issues/83706 because with current change this becomes more impactful as it is easier to get into this confusing state.

academo committed:
[Security Solution] host isolation exceptions listing under policy integration details tab (#120361)

flash1293 committed:
wait for vis before asserting on it (#121083)

nreese committed:
[maps] fix flaky test functional/apps/maps/embeddable/dashboard·js (#120771)

* unskip

* .only

* add retry around failing section

* waitForLayersToLoad

* remove only

* eslint

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

nreese committed:
[maps] fix test/functional/apps/maps/embeddable/tooltip_filter_actions·js (#120854)

* [maps] fix test/functional/apps/maps/embeddable/tooltip_filter_actions·js

* add method to wait for layers with collapsed layer control

* type

* unskip flaky test

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

nreese committed:
[maps] fix flaky test /functional/apps/maps/add_layer_panel·js (#120990)

* [maps] fix flaky test /functional/apps/maps/add_layer_panel·js

* eslint

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

semd committed:
[Cases] Fix flaky test on form submit (#121073)

* change to use simulate click for submitting

* lint

* type check

cauemarcondes committed:
[APM] Add comparision to service maps popover (#120839)

* adding comparison to service maps

* adding tests

* addressing pr comments

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
Some updates to our dev docs (#120981)

* Master -> main update branching strategy with make it minor

* Follow the new dev docs process and keep nav inside this repo

* add back nav links that are in a different repo

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

michaelolo24 committed:
[Security Solutio][Investigations] Update Timeline Details API with ECS field (#120683)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

kevinlog committed:
[Security Solution] Fix broken redirect links in between Fleet and Security (#120897)

Commit:
[Alerting] Consider expired tasks invalid (#119664)

* Consider expired tasks as invalid

* Since we do not reschedule expired tasks now, this test needs to change

* Add unit test

* Fixing functional test

* Fix types

* Fix bad merge

* Remove unused variable

* Better fix for bad merge

* Relax tests

* See if this is flaky

* Try fixing this another way

* Skip tests for now

Co-authored-by: Ying Mao <ying.mao@elastic.co>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

brianseeders committed:
[TypeScript] Upgrade to 4.5.3 (#120812)

dominiqueclarke committed:
[Fleet] index templates - add support for wildcard multi-fields (#121088)

* fleet - index templates - add support for wildcard multi-fields

* adjust types

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

jloleysens committed:
[Reporting] Close chromium before compiling pdf (#120223)

* added "close" method to the factory output so that consumers can close the browser once they are done with the page

* update test reference to exit$

* removed old files

* fix variable name

* remove unused import and fix typo in mock

* added comments and documentation for close functionality

* updated implemenatation of close to not call .kill if page is already closed

* added a test describing the close behaviour we want, also moved the page.isClosed() check to the private .kill function to be DRY

* avoid emitting a log unnecessarily

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

cauemarcondes committed:
[APM] Don't show deprecation notice in Kibana Upgrade Assistant for non-apm users (#120945)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

stephmilovic committed:
[Security Solution] [Sourcerer] Bug fixes for hanging load and modified label (#120814)

tylersmalley committed:
[Ironbank] Upstream updates to entryfile permissions (#121113)

nchaulet committed:
[Fleet] Fix increment package policy name (#121101)

Commit:
[failedTestsReporter] use ci-stats to find existing issues (#120875)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

lukasolson committed:
[kql] Remove named args and unused geo functions (#118973)

* [kql] Remove named args and unused geo functions

* Fix tests

* Re-format grammar output

brokensound77 committed:
[Detection Rules] Add 8.0 rules (#121045)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
chore(NA): splits types from code on @kbn/field-types (#121077)

afharo committed:
[PluginService] Register all deprecations before running `isEnabledPath` (#121084)

kqualters-elastic committed:
[Security Solution] Add app id prop for navigateToApp (#121139)

* Add app id prop for navigateToApp

* Add missing hook dependency

* Fix types

stephmilovic committed:
[Security Solution] Sourcerer Redesign Cypress tests (#118271)

estermv committed:
Remove unused files (#121104)

ymao1 committed:
[Alerting] Adding rule executor service functions to support execution cancellation (#120506)

* Adding abortable es client factory

* Starting to add tests

* Tests

* Fixing types

* Fixing types

* Fixing types

* Adding functional tests

* Reverting changes to rule type

* Handling the correct error message

* Cleanup

* Fixing preview rules route

* Updating README

* Removing unneeded observable

* Cleanup

* Fixing types

* Using abort signal

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

lucasfcosta committed:
[Uptime] Disable 'Create Rule' button when user doesn't have uptime write permissions [#118404] (#120379)

* [Uptime] Disable 'Create Rule' button when user doesn't have uptime write permissions [#118404]

Before this commit, users would be able to open the flyout to create an
alert and would end-up seeing an error toast when they tried to save it.

This commit will now disable the create alert button when the user
doesn't have permissions to write to Uptime. It will also display a
helpful tooltip.

* [Uptime] Disable "Enable Anomaly Alert" when users can't write to uptime [#118404]

This commit causes users not to be able to use the "Enable Anomaly
Alert" button within the popover in the monitors screen. That button
will now be disabled and contain an informative tooltip whenever users
don't have permissions to write to Uptime.

We've chosen to take this approach so that we don't have to modify the
component which deals with the alert creation, which belongs to another
team and that we plan on eventually replacing. Furthermore, this pattern
is already used in the logs app.

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dej611 committed:
[Lens] Revisit flyout fields style from column to row compressed (#120103)

* :lipstick: Revisit form style + spacers

* :white_check_mark: Fix tests

* :sparkles: Replace percentile input with range

* :lipstick: Fix mobile layout

* :lipstick: Make searchbar go below flyout

* :fire: Remove unused jsx

* [Lens] Refactor Flyout Design Updates (#14)

* allow flyout to go full width on small viewports

* remove hiding of layer panel when flyout open

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Michael Marcialis <michael@marcial.is>

mibragimov committed:
Fix autocomplete suggestions for lowercase methods and other related bug (#121033)

* Fix autocomplete suggestions for lowercase methods and another related bug

Co-authored-by: Muhammad Ibragimov <muhammad.ibragimov@elastic.co>

joshdover committed:
[Fleet] Fix typo in Fleet Server callout (#121070)

academo committed:
[Security Solution] host isolation exceptions generator by policy (#120490)

flash1293 committed:
Make sure saved search id is carried over to saved object (#121082)

juliaElastic committed:
[Fleet] fix bug when dryrun gave error on latest package policy version (#121094)

* fix bug when dryrun gave error on latest package policy version

* fix bug when package is older than policy

* added unit tests

* fix test

* fix tests

mbondyra committed:
[Lens] flaky gauge  fix (#120724)

* flaky fix

* flaky

* the other flakiness

* retrySetValue everywhere

* one option changed

* move to original order

* Update lens_page.ts

pgayvallet committed:
Improve SOR.find reference filter implementation (#121042)

mistic committed:
chore(NA): splits types from code on @kbn/mapbox-gl (#121078)

* chore(NA): splits types from code on @kbn/mapbox-gl

* chore(NA): fix exported types

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

juliaElastic committed:
added missing option to fleet server intructions (#121180)

thomheymann committed:
Fix setup CLI (#121124)

renovate[bot] committed:
Update dependency broadcast-channel to ^4.7.1 (#121155)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

brianseeders committed:
[CI] Create a pipeline to bootstrap and populate bazel remote cache for MacOS (#120136)

nchaulet committed:
[Fleet] fix redudant copy in output & settings confirm dialog (#121187)

loixlab committed:
[CCR] Unskip follower indices component integration test (#120912)

loixlab committed:
[ILM] Unskip timing validation (#120967)

marshallmain committed:
[RAC][Security Solution] Add flattened parameters object and populate it in Security Solution (#120698)

* Add flattend parameters object and populate it in Security Solution

* Fix severity, risk_score, bugs, tests

* Add ALERT_RULE_PARAMETERS to package

* Skip tightly coupled test

* fix more tests

* Remove unused import

* Fix threat matching API test

* Continue overriding kibana.alert.rule.risk_score and severity for now

* Add ignore_above to ALERT_RULE_PARAMETERS

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dedemorton committed:
Update Fleet overview topic (#121153)

qn895 committed:
[ML] Add Maps ui action to Index data visualizer/Discover's Field statistics (#120846)

* Add map UI actions to dv

* Fix maps logo, comment

* Add permission check for ui actions

* Remove unnecessary json stringify because it no longer includes query object

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

brianseeders committed:
Bump prettier/eslint packages related to Typescript after TS upgrade (#121119)

YulNaumenko committed:
[Alerting] Fixed Telemetry task: "Cannot read properties of undefined" due to the missing shard on Kibana start (#120438)

* [Alerting] Fixed Telemetry task: "Cannot read properties of undefined" due to the missing shard on Kibana start

* fixed update errors

* fixed tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

paul-tavares committed:
[Security Solution][Endpoint] Add explicit Authorization validations to Endpoint APIs (#120482)

* new higher order Request Handler for endpoint authz enforcement (withEndpointAuthz)
* correct isolation status permission

qn895 committed:
[ML] Add auto generated drill down link to Discover for Anomaly explorer table (#120450)

* Add auto generated drill down link to Discover

* Fix data view, convert to typescript, use format hooks

* Use kibana formatting

* Handle special characters

* Fix mlcategory/categorization detectors in query

* Fix to influencer

* Fix start/end interval logic

* No need to show Discover option when a categorization anomaly

* Fix timezone issue when timezone is not default browser

* Use AnomaliesTableRecord

* Remove unused Anomaly type

* Change to import type

claudiopro committed:
[Observability] Passes `theme$` to `toMountPoint` (#120968)

* Treats toMountPoint - part II

* Adds theme$ to exploratory view config

* Uses theme$ from useExploratoryView hook

* Uses theme$ in HeaderActionMenuContext

* Better props for MonitoringApp

vitaliidm committed:
[Security Solution][Detections]Fix rule query overlapping timeline banner on rule edit page #121127

jbudz committed:
[ci] Support building OS packages (#121146)

This adds support for building deb, rpm, and docker images using the
ci:build-os-packages label.

Depends on https://github.com/elastic/kibana-buildkite/pull/2

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[ci/flaky] support triggering a with config via the env (#121215)

qn895 committed:
Fix layer type name to new convention (#121218)

KOTungseth committed:
[DOCS] Updates {data-source} in Canvas (#121216)

KOTungseth committed:
Remove CentOS (#121128)

nreese committed:
[Maps] import Filter from @kbn/es-query (#120690)

* import Filter filter from @kbn/es-query

* eslint

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

gchaps committed:
[DOCS] Moves encryption key doc to tools section (#121021)

* [DOCS] Moves encryption key doc to tools section

* Update docs/user/commands/encryption-keys/index.asciidoc

Co-authored-by: Joe Portner <5295965+jportner@users.noreply.github.com>

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Joe Portner <5295965+jportner@users.noreply.github.com>

parkiino committed:
[Security Solution][Endpoint][Admin][Policy List] GET endpoint package policy api (#119545)

muneebsajjad committed:
[APM] Update "customize-app" route path (#121032)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Heenawter committed:
[Dashboard] Fix blank panel save and display issue. (#120815)

* Added special case to panel comparison.

* Fixed panel title behaviour when linking/unlinking to/from the library.

* Added panel title test suite.

* Made the injection of the 'savedSearchId' key conditional on if it is defined when unlinking.

* Adjusted use of 'setValue()' based on feedback from @dmlemeshko

* Refractor test suite.

* Improved code for removal of title on library link.

mbondyra committed:
[Lens] move gauge to chart expression plugin (#120623)

* gauge_expression

* fix

* CI fix

* make plugin required to use its methods

* move non-types from types

* share icon

* bundle, plugin

* load gaugeComponent async

* fix types

* async heatmap

* don't use d3

* optimizations

* feedback

* types

* adjusting the tests to the new max

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dasansol92 committed:
[Security Solution] [Endpoint] User can assign entry to one or more policies when creating an event filter (#120913)

* Adds effect scope form on event filters form

* Load package policies using react-query hook

* Fixes and adds unit tests

* Disables save button if policies are loading to prevent user save event filter without the policies loaded

* Fixes bug in form that didn't save the policies changes if something else was changed before save. Also adds policies loading state within the loader condition

* Fixes typo in comment

* Fixes multilang key

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dasansol92 committed:
Display's empty state when there are no assigned event filters to policy or not event filters exists at all (#121162)

janmonschke committed:
fix: show correct alert actions based on the alert's status (#121203)

jloleysens committed:
[Reporting] Reporting info panel touch ups (#120617)

* changed info -> summary and combined maximum attempts and attempts field

* added spaceId if it is there to job info

* moved space id to job details section

* moved processing and job info into a single array

* wip, introduced a new structure to the flyout for easier visual parsing. still need to re-introduce errors and deprecated report types in the new structure

* remove unnecessary div

* added warnings and errors to callout and cleaned up comments

* remove unused functionality and introduce date formatting per advanced settings format

* remove unused import

* remove unused i18n

* refactor check for info.max_attempts in flyout content

Co-authored-by: Tim Sullivan <tsullivan@users.noreply.github.com>

* update info.lyout.dimensions.height to be Math.ceiled

Co-authored-by: Tim Sullivan <tsullivan@users.noreply.github.com>

* update info.lyout.dimensions.width to be Math.ceiled

Co-authored-by: Tim Sullivan <tsullivan@users.noreply.github.com>

* lint and remove bad suggestion commit

* added actions button to flyout, in flyout footer

* create a little bit more breathing room for the section titles

* added logic to disable the action buttons in the flyout when the actions are not ready

* update copy per feedback

* added basic tests for the actions menu in the flyout

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Tim Sullivan <tsullivan@users.noreply.github.com>

Commit:
[Discover] Add error state if chart loading fails (#119289)

* [Discover] Add error state if chart loading fails

* Reorder stuff around

* Remove file

* Fix error message a bit

* Update src/plugins/discover/public/application/main/components/chart/histogram.tsx

Co-authored-by: Andrea Del Rio <delrio.andre@gmail.com>

* Add error handling

* Add a unit test

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Andrea Del Rio <delrio.andre@gmail.com>

kertal committed:
[Discover] Add i18n to no results screen (#121065)

dej611 committed:
[Lens] Add multiterms telemetry (#121178)

* :sparkles: Add multiterms telemetry

* :fire: remove duplicate data

* :wrench: Fix schema issue

* :ok_hand: Remove log

Kerry350 committed:
Replace IFieldType (#121108)

bdentino committed:
[Logs UI] Query numerator & denominator simultaneously for log threshold alerts (#107566)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

academo committed:
[Security Solution] host isolation exceptions filter list by policy (#120598)

paul-tavares committed:
[Security Solution][Hosts, Endpoint] Fix Host Details not showing endpoint related data for users without superuser role (#121090)

* add new method to EndpointAppContextServices for getting a internal fleet services interface
* Change Host Details search Strategy to use internal fleet services
* un-skip host details FTR test

nchaulet committed:
[Fleet] Do not udpate or rollover replicated datastream (#121235)

nchaulet committed:
[Fleet] Use Fleet authz for fleet server routes security instead of enforcing superuser (#119795)

jbudz committed:
[ci] Upload deb and rpm artifacts (#121228)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

juliaElastic committed:
[Fleet] added kibana version to user-agent (#121297)

* added kibana version to user-agent

* Update x-pack/plugins/fleet/server/services/epm/registry/requests.ts

Co-authored-by: Nicolas Chaulet <n.chaulet@gmail.com>

Co-authored-by: Nicolas Chaulet <n.chaulet@gmail.com>

juliaElastic committed:
added missing envvar (#121308)

semd committed:
[Cases] Case details UI KPI metrics (#119463)

* Working lifespan metrics api

* new case metrics container and api call

* Adding remaining metrics handlers and some tests

* Fixing jest snapshot

* Switch to kbn archiver

* tests added, case view page refactor

* test for metrics component added

* fix type

* fix responsivenes on small screens

* type fixes

* use new features prop for case metrics

* test fixed

* fix CasesFeatures type

* integration test fix

* changes and suggestions

* metrics features implementation and connectors type

Co-authored-by: Jonathan Buttner <jonathan.buttner@elastic.co>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
[Dasboard] Update embeddable rendering test for dashboard (#121000)

* Update embeddable rendering test on dashboard

* Updating dashboard fixture to remove old viz

alvarezmelissa87 committed:
Update layer type name to new convention (#121254)

nreese committed:
[Maps] fix drawings do not show when there is a global filter (#121239)

* [Maps] fix drawings do not show when there is a global filter

* i18n clean up

janmonschke committed:
[Security Solution][Investigations] Don't show the empty label in charts for number values (#121274)

* fix: don't show the empty label for number values

* test: improve test message

paul-tavares committed:
[Security Solution][Endpoint] Add authz check to /api/endpoint/policy API (#121312)

brianseeders committed:
[kbn-es] Allow ES snapshot version mismatch with Kibana (#120942)

brianseeders committed:
[renovate] Add Typescript configuration (#121245)

Commit:
Fetch rule actions in chunks (#121110)

KOTungseth committed:
[DOCS] Updates {data-source} in REST APIs (#121240)

* [DOCS] Updates {data-source} in REST APIs

* Update docs/api/saved-objects/bulk_get.asciidoc

Co-authored-by: Matthew Kime <matt@mattki.me>

Co-authored-by: Matthew Kime <matt@mattki.me>

KOTungseth committed:
[DOCS] Updates {data-source} in Dashboard and visualizations (#121213)

Heenawter committed:
Removed edit menu when in view only mode. (#118779)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

nreese committed:
[Maps] Remove redundant categories from field-meta type (#121338)

* [Maps] Remove redundant categories from field-meta type

* clean up another place where null is returned

* fix jest test

TinaHeiligers committed:
Adds event loop delay percentiles to mean delay (#121052)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

gchaps committed:
[DOCS] Fixes inconsistency with role management link (#121351)

* [DOCS] Fixes inconsistency with role management link

* Update docs/redirects.asciidoc

Co-authored-by: Lisa Cawley <lcawley@elastic.co>

Co-authored-by: Lisa Cawley <lcawley@elastic.co>

Commit:
Migrated ILM Tests to Use test_user (#121262)

* Updated ILM tests to use test user and updated permissions.

* Updated security tests to check for Advanced Settings in the navigatio pane since we added it to the Kibana permissions.

kqualters-elastic committed:
[Security Solution] Alerts visualization free field selection (#120610)

* Use EuiComboBox in place of a select for alert page visuals

* Remove console.log

* Better variable name in reduce function

* Remove comment

* Remove typo

* Fix linting/tests

* PR feedback

* Add missing mocks

* Add router mocks

* Rename getAggregatableFields

Bamieh committed:
[Usage Collection] add caching layer for stats (#119312)

kqualters-elastic committed:
Don't show page title on empty view (#121350)

dedemorton committed:
Fix link to docs about on prem registry (#121236)

matschaffer committed:
[Stack Monitoring] Use KibanaThemeProvider in the Stack Monitoring UI toast notifications (#120713)

This PR originally added the KibanaThemeProvider to stack monitoring as well, but https://github.com/elastic/kibana/pull/120968 was opened to do the same thing.

Instead this change focuses on updating the toMountPoint calls found in stack monitoring toast notifications to ensure they're either handled by simple i18n calls or appropriately pass a theme$ argument when called.

mbondyra committed:
[Lens] add ticks on bands tooltip for gauge (#121304)

* [Lens] add ticks on bands tooltip for gauge

* Update x-pack/plugins/lens/public/visualizations/gauge/dimension_editor.tsx

Co-authored-by: Marco Liberati <dej611@users.noreply.github.com>

Co-authored-by: Marco Liberati <dej611@users.noreply.github.com>

gbamparop committed:
[APM] Add documentation for the agent keys API in the APM app (#120787)

* Add documentation for the agent keys API in the APM app

Co-authored-by: Brandon Morelli <bmorelli25@gmail.com>
Co-authored-by: Søren Louv-Jansen <sorenlouv@gmail.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

peteharverson committed:
[ML] Data visualizer: Use KibanaThemeProvider (#121286)

* [ML] Data visualizer: Use KibanaThemeProvider

* [ML] Edits following review

dasansol92 committed:
Update codeowners paths for OLM team (#121319)

mbondyra committed:
[Lens] fix limits to format selector (#121294)

* test

* flatten the structure

* fix test

dimaanj committed:
[Discover] Fix warning of Document explorer header cell (#121071)

* [Discover] fix warning

* [Discover] update test

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mibragimov committed:
[Console] Add links to docs (#121066)

* Add links to docs

Co-authored-by: Muhammad Ibragimov <muhammad.ibragimov@elastic.co>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Dosant committed:
fix field formats integration in field editor (#121300)

dasansol92 committed:
[Security Solution] [Endpoint] Allow filter by policy in event filters list (#121305)

* Allow filter by policy in event filters list

* Fix ts type errors

* Make exclude policies optional and fix unit test using userEvent

thomheymann committed:
Detect cookies on login page (#120944)

Commit:
[Graph] Paginate through index patterns (#120972)

* [Graph] Paginate through index patterns

* Extract logic into a generator function

* Fix failing unit test

* Fix page initialization

* Update x-pack/plugins/graph/public/helpers/use_workspace_loader.ts

Co-authored-by: Matthias Wilhelm <ankertal@gmail.com>

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Matthias Wilhelm <ankertal@gmail.com>

nchaulet committed:
[Fleet] Fix add agent authentication settings display when no tokens (#121356)

xcrzx committed:
Test APM instrumentation of rule executors (#117672)

mistic committed:
chore(NA): adds backport configs for 7.17 branch (#121398)

Dosant committed:
fix <CodeEditor/> font glitches (#121392)

criamico committed:
[Fleet] Update the openApi readme section (#121331)

semd committed:
[Cases] Activate Case detail metrics (#121336)

* activate metrics for secsol and o11y

* add connectors to o11y

* disable o11y metrics

Kerry350 committed:
[Stack Monitoring] Fix date picker range options  (#121295)

* Pass commonly used ranges to EuiSuperDatePicker

* Tidy up start contract types

mshustov committed:
Document execution_context feature (#120894)

* document execution_context

* Apply suggestions from code review

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>
Co-authored-by: Luke Elmers <lukeelmers@gmail.com>
Co-authored-by: Christiane (Tina) Heiligers <christiane.heiligers@elastic.co>

* format log output

* restructure pages

* Apply suggestions from code review

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>
Co-authored-by: Luke Elmers <lukeelmers@gmail.com>
Co-authored-by: Christiane (Tina) Heiligers <christiane.heiligers@elastic.co>

criamico committed:
[Fleet] Add missing endpoint to openApi docs (#121091)

* [Fleet] Add missing endpoint to openApi docs

* Fix parsing errors

* Add deprecations and fix other errors

* Fix validation errors and address code review comments

* Add Update EPM packages endpoint

* Mark clearly deprecations in response parameters

* Add missing requestBody and address other comment

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mbondyra committed:
[Lens] don't update searchSession on start (#121385)

* [Lens] don't update searchSession on start

* Update index.ts

* Update index.ts

ashokaditya committed:
[Security Solution][Endpoint] Improve endpoint performance with TA wildcard paths (#120349)

* Show full executable names in placeholder for wildcard paths

fixes elastic/security-team/issues/2293

* Show soft warning when wildcard also in executable names

fixes elastic/security-team/issues/2293

* add wildcard path entries to fake TA list

refs elastic/security-team/issues/2293

* Append a process.name entry when executable name in wildcard path

fixes elastic/security-team/issues/2293

* commit using ashokaditya@elastic.co

* linux should always use  `_cased` types

review changes

* use better TS

* use matcher functions to compute operator value for linux

review suggestions

* use path to extract filenames on server side

review suggestions

* improve regex for windows and unix filepaths

review suggestions

* update test mocks

review changes

* update regex to match multi spaces and single chars with spaces in filenames

* add comment to explain

review suggestions

* update copy

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

tsullivan committed:
[Reporting] Clarify deprecation changes of the roles settings (#121349)

* [Reporting] Clarify deprecation changes of the roles settings

* updates per feedback

* update test snapshots

* minor fix

* fix i18n

* remove formatting marks

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

vadimkibana committed:
Merge branch 'fix-ctx-drill' of github.com:vadimkibana/kibana into fix-ctx-drill

vadimkibana committed:
always show trigger picker

prakashgupta1605 commented:
I am trying to add a dashboard drill for a lens metrics. It let's me add the dashboard link but the visualization is not clickable. I am using Kibana 8.10 version. Could someone tell me if I am missing some setting or this issue has introduced again?

nickpeihl commented:
> I am trying to add a dashboard drill for a lens metrics. It let's me add the dashboard link but the visualization is not clickable. I am using Kibana 8.10 version. Could someone tell me if I am missing some setting or this issue has introduced again?

@prakashgupta1605 Can you post your question in https://discuss.elastic.co/c/elastic-stack/kibana/7 and include screenshots? We prefer to keep questions in Discuss because it has a wider audience which means your question is much more likely to get a helpful response.
